parcel-resolver-veda/index.d.ts

```diff
@@ -62,13 +62,27 @@ declare module 'veda' {
   // resolved from DatasetLayerCompareSTAC or DatasetLayerCompareInternal. The
   // difference with a "base" dataset layer is not having a name and
   // description.
-  export interface DatasetLayerCompareNormalized
+  export interface DatasetLayerCompareBase
     extends DatasetLayerCommonCompareProps {
     id: string;
     stacCol: string;
     type: DatasetLayerType;
   }
 
+  interface DatasetLayerCompareNoLegend extends DatasetLayerCompareBase {
+    legend?: never;
+    name?: never;
+    description?: never;
+  }
+      
+  interface DatasetLayerCompareWLegend extends DatasetLayerCompareBase {
+    legend: LayerLegendCategorical | LayerLegendGradient | undefined;
+    name: string;
+    description: string;
+  }
+
+  export type DatasetLayerCompareNormalized = DatasetLayerCompareNoLegend | DatasetLayerCompareWLegend
+
```

Comment:
Perhaps it is not needed to create types for compare layers with and without legend.

After the compare layer is "normalized" it should have a legend since we need to display it.
```tsx
  export interface DatasetLayerCompareNormalized
    extends DatasetLayerCommonCompareProps {
    id: string;
    stacCol: string;
    type: DatasetLayerType;
    name: string;
    description: string;
    legend?: LayerLegendCategorical | LayerLegendGradient;
  }
```

This is ok when using a internal layer to compare (by providing a dataset and layer id), since the defined legend will be gotten from that layer.
When using a layer with a STAC definition (`DatasetLayerCompareSTAC`), we need to start requiring a legend, since it is not defined anywhere else.
```tsx
  export interface DatasetLayerCompareSTAC
    extends DatasetLayerCommonCompareProps {
    stacCol: string;
    type: DatasetLayerType;
    name: string;
    description: string;
    legend?: LayerLegendCategorical | LayerLegendGradient;
  }
```

Reply:
ah I was thinking in a way that if comparelayer is fetching the layer from stac endpoint, there is no way to know about the legend but I guess we should ask in this case. Thank you for 💭 !